module/programs/helper.js

```diff
@@ -1019,6 +1020,120 @@ module.exports = class ProgramsHelper {
     });
   }
 
+  /**
+   * Program details or cound o programs based on user.
+   * @method
+   * @name userProgram
+   * @param {String} reqesteduserId - userId .
+   * @param {String} stats          - condition to check stats or details .
+   * @returns {Object}              - Details of the program or count of the program.
+   */
+
+  static userProgram(reqesteduserId, stats = true) {
+    return new Promise(async (resolve, reject) => {
+      try {
+        if (stats === false) {
+          //getting details of usersProgram and programJoined details
+
+          let updateUserProgramList = await this.getAggregate(reqesteduserId);
+          return resolve({
+            message: constants.apiResponses.PROGRAMS_FETCHED,
+            success: true,
+            data: updateUserProgramList,
+          });
+        } else {
+          //getting bigNumber of usersProgram list
+          let countProgram = await programUsersHelper.countDocuments({
+            userId: reqesteduserId,
+          });
+          return resolve({
+            message: constants.apiResponses.PROGRAMS_FETCHED,
+            success: true,
+            data: countProgram,
+          });
+        }
+      } catch (error) {
+        return reject({
+          success: false,
+          status: error.status
+            ? error.status
+            : httpStatusCode["internal_server_error"].status,
+          message: error.message,
+        });
+      }
+    });
+  }
+
+  /**
+   * aggregate function.
+   * @method
+   * @name getAggregate
+   * @param {String} [createdBy]         - userId
+   * @returns {Array}                    -program details.
+   */
+
+  static getAggregate(createdBy) {
```

Comment:
reverse the query @praveenKDass you are running query on program it should be on programUser

Reply:
Reversed the query @ankit1shahu